### PR TITLE
Non used variable & file deletion

### PR DESCRIPTION
**Pull request by Lee-mk-Joseph**
Removed non used variables from variables.tf & locals.tf
Transfered the 3 x data blocks over to 'data.tf' and deleted the 'VPC.tf' file to promote consistency
Very similar actions taken to that of the completed work in DVOP-2496:
https://companieshouse.atlassian.net/browse/DVOP-2496
This PR is covered via DVOP-2547:
https://companieshouse.atlassian.net/browse/DVOP-2547

### CODE REVIEW

File: groups/rds/data.tf [hunk 1]
```diff
@@ -6,6 +6,7 @@ data "vault_generic_secret" "secrets" {
   path = "team-${var.team}/${var.account_name}/${var.region}/${var.environment}/${var.repository_name}"
 }
 
+data "aws_region" "current" {}
```
Comment (sienkin) [inline groups/rds/data.tf:9]: This data resource doesn't appear to be used anywhere else - is it required?

File: groups/rds/locals.tf [hunk 2]
```diff
@@ -13,20 +13,10 @@ locals {
...
-  automation_subnet_cidrs = values(zipmap(
-    values(data.aws_subnet.automation).*.availability_zone,
-    values(data.aws_subnet.automation).*.cidr_block
-  ))
-  automation_subnet_ids = values(zipmap(
-    values(data.aws_subnet.automation).*.availability_zone,
-    values(data.aws_subnet.automation).*.id
-  ))
   automation_subnet_pattern = local.secrets.automation_subnet_pattern
   automation_vpc_pattern    = local.secrets.automation_vpc_pattern
```
Comment (sienkin) [inline groups/rds/locals.tf:17]: Is this required? Typically the specification of the automation subnets would be if Concourse access is required, which is unlikely to be the case here.

File: groups/rds/data.tf (hunk 1 above)
Comment (Lee-mk-Joseph) [reply groups/rds/data.tf:9]: Does not be not be used by anywhere. Removed from both instance & RDS groups

File: groups/rds/locals.tf (hunk 2 above)
Comment (Lee-mk-Joseph) [reply groups/rds/locals.tf:17]: Automation subnet & subnets not required. The following have been removed.

**local.tf:** automation_vpc_pattern & automation_subnet_pattern
**data.tf:** "aws_vpc""automation", "aws_subnet_ids""automation", "aws_subnet""automation"

**Review by Ngriffiths1 — APPROVED** [review]

**Pull request merged**